### ISSUE TITLE
Monitor log is flooded by a warning.

### ISSUE THREAD

**Issue by giuseppe-carboni** [discos/srt-single-dish-tools#150]
The warning that fills the monitor log is the following.
`/lib/python3.6/site-packages/numpy/core/fromnumeric.py:688: UserWarning: Warning: 'partition' will ignore the 'mask' of the MaskedColumn.
  a.partition(kth, axis=axis, kind=kind, order=order)`
With the logfile flooded by it it's difficult to get any more information about what's going on under the hood. I believe that it's raised by some astropy/numpy call, but I don't really know where to look.
@matteobachetti is there any chance you can help me with this one?

**Comment by matteobachetti** [comment]
Hi Giuseppe,

somewhere, I'm calculating the median of a masked array. Here is the commit log where the warning is introduced, and it says exactly that.
I'll look for the culprit :)
```
portabile2:numpy $ git log -S"will ignore the " -p 
commit d0bf15d1b99dd239530b25a0d939ee4475f85af5
Author: Michael Seifert <michaelseifert04@yahoo.de>
Date:   Wed Feb 22 16:50:49 2017 +0100

    MAINT: Warn users when calling np.ma.MaskedArray.(arg-)partition function.
    
    Using the np.median function on MaskedArrays uses the not-overriden
    partition method of a plain np.ndarray without error or warning. (#7330)
    This PR overrides the partition method on MaskedArrays but simply to
    throw a Warning. This will make users aware that something ignores the
    mask without breaking backwards-compatibility. This also applies to
    the argpartition method (even if it's not called by np.median).

diff --git a/numpy/ma/core.py b/numpy/ma/core.py
index af8a523ab..ea4a1d85f 100644
--- a/numpy/ma/core.py
+++ b/numpy/ma/core.py
@@ -5635,6 +5635,18 @@ class MaskedArray(ndarray):
         np.subtract(out, min_value, out=out, casting='unsafe')
         return out
 
+    def partition(self, *args, **kwargs):
+        warnings.warn("Warning: 'partition' will ignore the 'mask' "
+                      "of the {}.".format(self.__class__.__name__),
+                      stacklevel=2)
+        return super(MaskedArray, self).partition(*args, **kwargs)
+
+    def argpartition(self, *args, **kwargs):
+        warnings.warn("Warning: 'argpartition' will ignore the 'mask' "
+                      "of the {}.".format(self.__class__.__name__),
+                      stacklevel=2)
+        return super(MaskedArray, self).argpartition(*args, **kwargs)
+

```

**Comment by matteobachetti** [comment]
In the meantime, would something like this work?
```
import warnings
with warnings.catch_warnings():
    warnings.filterwarnings("once", message="divide by zero encountered in divide")
    main_process(... etc ...)
```
I don't have the warning in my computer, so I don't know how to test this.

**Comment by giuseppe-carboni** [comment]
This was fixed in #163, closing.
